docs/deploy-on-k8s.md

```diff
@@ -0,0 +1,115 @@
+
+As this cache uses the disk, which is not easily shared amongst pods, this also means the API Statefulsets cannot be safely scaled. If there were to be more than one main-api pod, for example, each pod would see different order creations, and therefore have a different cache of orders. This would result in unpredictable behaviour.
+
+These are all limitations of our deliberately naeive caching mechanism, and would not occur should the APIs use a shared database such as MySQL or Postgres.
+
+## Using the application
+
+So now you have the application installed in Kubernetes, how can you use it? The standard practise for accessing a service inside a Kubernetes cluster would be to create a Service with type LoadBalancer. On AWS, for example, this would cause AWS to automatically provision an ELB or ALB for you and connect it to the Kubernetes service, allowing external access. As we are only providing a demonstration system and don't wish to make things available over the public internet, we are not using LoadBalancer in our deployment. Instead we provide some scripts which use Kubernetes port-forwarding so that you can reach the Services in Kubernetes from your local machine.
+
+To see the application's Web UI you can run the script:
+
+```sh
+./deployments/port-forward-web.sh
+```
+
+That command will continue to run until you kill it (with Ctrl+C) for example. While the command is running it will forward any traffic from your localmachine on port 3000 to the application's Web UI in the cluster. You should now be able to see the application at: http://localhost:3000/
+
+We also provide two other port forwarding scripts. The first is to give you access to the web UI for the Temporal cluster running in your cluster:
+
+```sh
+./deployments/port-forward-temporal-ui.sh
+```
+
+While this command is running you will be able to see the Temporal Web UI at: http://localhost:8080. Try placing an order with the application's Web UI and then looking at the Temporal Web UI to see which workflows were created.
+
+You may notice on the Temporal Web UI that the inputs and results for the Workflows and Activities are encrypted, due to the application's use of an encrypting data converter. In order to decrypt those on the Web UI so that you can see what the inptus and results were, you can use a codec server which we deployed alongside the app.
```

Comment:
```suggestion
You may notice on the Temporal Web UI that the inputs and results for the Workflows and Activities are encrypted, due to the application's use of an encrypting data converter. In order to decrypt those on the Web UI so that you can see what the inputs and results were, you can use a codec server which we deployed alongside the app.
```